File: First non-repeating character in a stream[17-July-2023]/Solution.py
#   >> PYTHON CODE <<

class Solution:
    def FirstNonRepeating(self, A):
        n = len(A)
        f = [0] * 26
        first = [-1] * 26
        
        for i in range(n):
            if first[ord(A[i]) - ord('a')] == -1:
                first[ord(A[i]) - ord('a')] = i
        
        ans = ""
        for i in range(n):
            f[ord(A[i]) - ord('a')] += 1
            ch = '#'
            x = n + 1
            for j in range(26):
                if f[j] == 1 and x > first[j]:
                    ch = chr(j + ord('a'))
                    x = first[j]
            
            ans += ch
        
        return ans

File: First non-repeating character in a stream[17-July-2023]/Solution.cpp
//  >>  CPP CODE <<

class Solution {
	public:
		string FirstNonRepeating(string A){
		    int n = A.size();
            vector<int> f(26, 0);
            vector<int> first(26, -1);
    
            string ans = "";
            for (int i = 0; i < n; i++) {

            	if (first[A[i] - 'a'] == -1) {
                    first[A[i] - 'a'] = i;
                }

                f[A[i] - 'a'] += 1;
                char ch = '#';
                int x = n + 1;
                for (int j = 0; j < 26; j++) {
                    if (f[j] == 1 && x > first[j]) {
                        ch = j + 'a';
                        x = first[j];
                    }
                }
    
                ans += ch;
            }
    
            return ans;
	    }

};
